File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    words = ["aya", "ye", "woo", "ma"]
    answer = 0

    for s in babbling:
        pos = 0
        last = ""
        valid = True

        while pos < len(s):
            matched = False
            for w in words:
                if s.startswith(w, pos) and w != last:
                    pos += len(w)
                    last = w
                    matched = True
                    break
            if not matched:
                valid = False
                break

        if valid:
            answer += 1

    return answer

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    const words = ["aya", "ye", "woo", "ma"];
    let answer = 0;

    for (let s of babbling) {
        let pos = 0;
        let last = ""; // 직전에 사용한 단어
        let valid = true;

        while (pos < s.length) {
            let matched = false;

            for (let w of words) {
                if (s.startsWith(w, pos) && w !== last) {
                    pos += w.length;
                    last = w;
                    matched = true;
                    break;
                }
            }

            if (!matched) {
                valid = false;
                break;
            }
        }

        if (valid) answer++;
    }

    return answer;
}
